### PR TITLE
fix(oauth): set TTL for the authorization request

### LINKED ISSUE #3982: Request settings page OAuth tokens with a TTL of 5 minutes
This is to help with #3928 where users can still interact with the settings page after they have been disconnected from another device.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3982

### CODE REVIEW

**Comment by vladikoff** [comment]
@shane-tomlinson or @seanmonstar or @vbudhram r?

**Comment by seanmonstar** [comment]
What happens if the user takes longer than 5 minutes on the settings page (such as using another tab for a bit, and then coming back)? Does it just error, or does it fetch a new token?

**Comment by vladikoff** [comment]
>  Does it just error, or does it fetch a new token?

Seems to work just fine, display name, avatar, basket, etc. 

Moved the `ttl` logic into `getToken` only when response type is `token`. 

cc @shane-tomlinson

File: app/scripts/lib/oauth-client.js [hunk 1]
```diff
@@ -39,7 +47,19 @@ define(function (require, exports, module) {
...
+     * @param {String} params.assertion
+     * @param {String} params.client_id
+     * @param {String} params.scope
+     * @returns {Promise}
+     */
+    getToken: function (params = {}) {
+      // set authorization TTL to 5 minutes.
+      // Docs: github.com/mozilla/fxa-oauth-server/blob/master/docs/api.md#post-v1authorization
+      // Issue: #3982
+      params.ttl = '300';
```
Comment (vbudhram) [inline app/scripts/lib/oauth-client.js]: @vladikoff Thoughts on making this configurable via [params](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/models/account.js#L216)? Maybe overkill?

Comment (vladikoff) [inline app/scripts/lib/oauth-client.js]: Maybe!

**Comment by shane-tomlinson** [comment]
> What happens if the user takes longer than 5 minutes on the settings page

OAuth tokens are [automatically re-provisioned if the current one has expired](https://github.com/mozilla/fxa-content-server/blob/cd98189e6c3307ee69fdbd60c0c4ceaa4aa83603/app/scripts/models/account.js#L849:L852).

File: app/scripts/lib/oauth-client.js (hunk 1 above)
Comment (shane-tomlinson) [inline app/scripts/lib/oauth-client.js]: > Thoughts on making this configurable via params, with default set to 300? Maybe overkill?

Seems totally reasonable.

**Comment by shane-tomlinson** [comment]
Other than the one comment about only setting TTL if not already set, looks great! Back to you to update and merge when ready.

**Pull request merged**